packages/core/src/diff/node.rs

```diff
@@ -402,9 +402,7 @@ impl VNode {
     }
 
     fn templates_are_different(&self, other: &VNode) -> bool {
-        let self_node_name = self.template.id();
-        let other_node_name = other.template.id();
-        self_node_name != other_node_name
+        self.template == other.template
```

Comment:
This function got inverted. It now checks if they are the same. I'm not sure where it is used